src/ATen/function_wrapper.py

```diff
@@ -311,37 +350,51 @@ def process_option(option, output_options):
         if re.match(EXCLUDE_PATTERN, option['name']):
             print("Excluding {}".format(option['name']))
             raise NYIError("NYI")
+
         # print(yaml.dump(option))
         formals = get_formals(option)
         option['formals_list'] = formals
         option['formals'] = [format_formal(f) for f in formals]
         option['returns'] = get_return_types(option)
+        option['return_type'] = format_return_type(option['returns'])
+        option['return_call'] = 'return ' if option['return_type'] != 'void' else ''
         option['actuals'] = [f['name'] for f in formals]
 
         option['method_formals'] = [format_formal(f) for f in formals
                                     if f['name'] != 'self']
         option['method_actuals'] = [
             f['name'] if f['name'] != 'self' else '*this' for f in formals]
-        option['return_type'] = format_return_type(option['returns'])
 
         option['const_mark'] = '' if option['inplace'] else ' const'
 
         is_method = 'method' in option['variants']
         is_function = 'function' in option['variants']
+        first_tensor = find_first_tensor(formals)
+        is_namespace_function = is_function and first_tensor is not None
```

Comment:
Similarly, what are the concepts of namespaces that we have introduced here?

Reply:
is_method = `Tensor::function_name()` exists
is_function = `Type::function()` exists and is "public" i.e. not just a `m_` prefix
is_namespace_function = accessible via `at::function()`